homeassistant/components/sunsynkweb/sensor.py

```diff
@@ -0,0 +1,177 @@
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_load",
+        translation_key="acc_load",
+        suggested_display_precision=2,
+    ),
+    SunsynkEntitySensorDescription(  # ""Total energy imported from the grid."""
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_grid_import",
+        translation_key="acc_grid_import",
+        suggested_display_precision=2,
+        agg_func=max,
+    ),
+    SunsynkEntitySensorDescription(  # ""Total energy exported to the grid."""
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_grid_export",
+        translation_key="acc_grid_export",
+        suggested_display_precision=2,
+        agg_func=max,
+    ),
+    SunsynkEntitySensorDescription(  # ""Total energy injected into the batteries."""
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_battery_charge",
+        translation_key="acc_battery_charge",
+        suggested_display_precision=2,
+    ),
+    SunsynkEntitySensorDescription(  # ""Total energy provided by the batteries."""
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_battery_discharge",
+        translation_key="acc_battery_discharge",
+        suggested_display_precision=2,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SunsynkConfigEntry,
+    async_add_entities: AddEntitiesCallback,  # noqa: F821
+) -> None:
+    """Set up sensor devices."""
+    coordinator: SunsynkUpdateCoordinator = entry.runtime_data
+
+    async_add_entities(
+        SunSynkApiSensor(coordinator, description)
+        for description in SENSOR_DESCRIPTIONS
+    )
+
+
+class SunSynkApiSensor(CoordinatorEntity[SunsynkUpdateCoordinator], SensorEntity):
+    """Parent class for sunsynk api exposing sensors.
+
+    The sensors expose the sum of the power across all inverters
+    for plants that have more than one inverter.
+    State of charge is normally shared between inverters, so this will take
+    the maximum of the state of charge.
+    """
+
+    _attr_has_entity_name = True
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Publish a device info."""
+        return DeviceInfo(
+            identifiers={
+                (DOMAIN, str(sum(plant.id for plant in self.coordinator.cache.plants))),
```

Comment:
why are you summing across plants here? I thought only inverters per plant get accumulated? I'm only noticing that now, but wouldn't it make more sense to add one device per plant?

Reply:
adding one device per plant requires the user 

* to create synthetic sensors to aggregate them up to get an overview 
* to know which sensors to sum up and which one will need a different aggregation function

both of which are not trivial. I've used a packet capture-based system to integrate it in my HA before via MQTT (before the packet format changed and I didn't want to reanalyse it) and having the multiple sensors per plant made it hard to integrate with the energy dashboard, which would be one of the main use cases.

This is very much an edge case as I expect the vast majority of domestic installations to be single inverter.

Reply:
to add to this, logically a plant should be able to be multiple inverters, and then yes a device per plant makes sense.
It just isn't how the data is structured by the service/api.

Reply:
in future versions, we might create one device per plant, one per inverter, one per PV string etc. and model everything separately. I think keeping things simple initially is a good starting point.

Reply:
I'm not sure I can follow your last addition - sorry. To me it looks like the data is already structured per plant in the cache?
The energy dashboards supports multiple input entities, so that's not a problem. I'm with you on the aggregate being non-trivial, but otherwise we take away the possiblity to look at them separately...

Reply:
I've never personally seen it useful to look at them separately, part of my objectives with the integration is to get a unified view, which none of the vendor's app give. Another part is to automate setting some of the system based on inputs from electricity provider re. current prices etc.
If the integration must not aggregate things like current power, it becomes a lot less useful to me personally.
What I'd suggest is that if there are other users of the integration that request Split View, we add that then.